src/transformers/models/pegasus/modeling_pegasus.py

```diff
@@ -1676,12 +1676,19 @@ def prepare_inputs_for_generation(
 
             input_ids = input_ids[:, remove_prefix_length:]
         # first step, decoder_cached_states are empty
-        return {
-            "input_ids": input_ids,  # encoder_outputs is defined. input_ids not needed
-            "attention_mask": attention_mask,
-            "past_key_values": past_key_values,
-            "use_cache": use_cache,
-        }
+        if inputs_embeds is not None and past_key_values is None:
+            model_inputs = {"inputs_embeds": inputs_embeds}
+        else:
+            model_inputs = {"input_ids": input_ids.contiguous()}
```

Comment:
Interesting - what's the reason for needing to add `.contiguous` here?

Reply:
Not much of a reason right now, as it is part of `compile` compatibilty. Decided to leave in case these models will be reworked for `compile` also